sw/device/tests/pwm_smoketest.c

```diff
@@ -0,0 +1,99 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_gpio.h"
+#include "sw/device/lib/dif/dif_pinmux.h"
+#include "sw/device/lib/dif/dif_pwm.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+#include "sw/device/lib/testing/test_framework/ottf_utils.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+static volatile const uint8_t kClocksHz[] = {24, 48, 96};
+static volatile const uint8_t kDutyCycles[] = {11, 36, 50, 71, 91};
+
+static dif_pwm_config_t compute_clk_config(uint32_t pwm_clk) {
+  enum {
+    kDutyCycleResolution = 4,
+    kBeatsPerCycle = 2 << (kDutyCycleResolution + 1),  // 2 ^ (DC_RESN + 1)
```

Comment:
The comment and code here are inconsistent; it should be `1 << (kDutyCycleResolution + 1)` or `2 << kDutyCycleResolution` for the same semantics as `dc_resn` in the HW IP documentation/register API. The behavior of the code is unaffected if you change '4' to '5' because the `...Resolution` value is not used elsewhere directly, but this would aid comprehension.